jmx-metrics/src/integrationTest/java/io/opentelemetry/contrib/jmxmetrics/AbstractIntegrationTest.java

```diff
@@ -206,6 +221,34 @@ private static void assertTypedPoints(List<NumberDataPoint> points, List<String>
                 .toArray(Consumer[]::new));
   }
 
+  @SuppressWarnings("unchecked")
+  private static void assertAttributedPoints(
+      List<NumberDataPoint> points, List<Map<String, String>> attributeGroups) {
+    assertThat(points)
+        .satisfiesExactlyInAnyOrder(
+            attributeGroups.stream()
+                .map(
+                    attributeGroup ->
```

Comment:
This is a bit too intense of an assertion to understand easily. Is it just checking that every point contains every attributeGroup? How about just passing in a map of merged attributes instead of separate groups?

Reply:
This is providing a list of the labelled data points expected, and validating that all of the expected combinations exist. For example, if we expected one data point with { operation: Read, status: Timeout } and one data point with { operation Read, status: Ok }, the attributeGroups parameter would contain a list of those two maps. Are there ways I could make that more clear?

Reply:
I think the complain is on the use of `.stream()` when you already have varargs to handle it.

E.g. I *think* you could do:

```
 assertThat(points)
        .extracting(point -> point.getAttributesList())
        .satisfiesExactlyInAnyOrder(
           list -> assertThat(list).contains(...),
           list -> assertThat(list).contains(...),
        );
```

Would also help if there was a "assertThatAttributeList(list, KV)" type helper method.

You can look in https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics-testing/src/main/java/io/opentelemetry/sdk/testing/assertj/metrics/MetricAssertions.java for ideas.  Might be nice to have a good set of assertion libraries for the OTLP protos...

Reply:
> and validating that all of the expected combinations exist

I may not be understanding well but currently think this is the key point. In particular the word "all". If so, it seems the parameter passed to the assertion could have all of the attributes in one Map rather than a list of maps. Is this not the case?

Reply:
Each item in the provided list represents one data point's AttributeList, and this method currently validates that we have N different data points that match N unique sets of AttributeLists. I don't see a way to reduce this to a single map. 

To expand on the example for this cassandra metric, we have 12 data points under one metric that we are validating, where each data point has a different combination of "operation" and "status" attributes. So the list contains the maps `{ operation: Read, status: Ok }, { operation: Read, status: Timeout }, { operation: Read, status: Unavailable }, { operation: Read, status: Failure }, { operation: Write, status: Ok }, { operation: Write, status: Timeout }, { operation: Write, status: Unavailable }, { operation: Write, status: Timeout }, { operation: Write, status: Failure }, { operation: RangeSlice, status: Ok },...`